### PR TITLE
feat(l1): add P2P Eth get receipts msg

### LINKED ISSUE #386: GetReceipts (0x0f)

### LINKED ISSUE #387: Receipts (0x10)

### PR DESCRIPTION
**Pull request by fkrause98**
**Motivation**

- Support the P2P eth functionality of exchanging transaction receipts.

Closes #386 
Closes #387

### CODE REVIEW

File: crates/common/types/receipt.rs [hunk 1]
```diff
@@ -31,6 +31,36 @@ impl Receipt {
...
+    // First, `inner_encode_receipt` will encode a receipt according
+    // to the RLP of its fields, if typed, the RLP of the fields
+    // is padded with the byte representing this type.
+    // For P2P messages, receipts are re-encoded as bytes
+    // (see the `encode` implementation for receipt).
+    // For debug and computing receipt roots, the expected
+    // RLP encodings are the ones returned by `inner_encode_receipt`.
+    // On some documentations, this is also called the `consensus-encoding`
+    // for a receipt.
+    pub fn inner_encode_receipt(&self) -> Vec<u8> {
```
Comment (fmoletta) [inline crates/common/types/receipt.rs:47]: Isn't `_receipt` redundant here? This is already a `Receipt` method

File: crates/common/types/receipt.rs [hunk 2]
```diff
@@ -45,55 +75,77 @@ fn bloom_from_logs(logs: &[Log]) -> Bloom {
...
+            }
+        };
     }
 }
 
 impl RLPDecode for Receipt {
+    /// Receipts can be encoded in the following formats:
+    /// A) Legacy receipts: rlp(LegacyTransaction)
+    /// B) Non legacy receipts: rlp(Bytes(tx_type | rlp(receipt))).
     fn decode_unfinished(rlp: &[u8]) -> Result<(Self, &[u8]), RLPDecodeError> {
```
[the remark this answers is not in the document]
Comment (fmoletta) [reply crates/common/types/receipt.rs:100]: The `RLPDecode` implementation is consistent with the implementation of `RLPEncode`, the if is to differentiate between legacy and non legacy transactions and not between the different encodings (RLPEncode vs inner_encode). I would add a an `inner_decode` that matches the inner_encode (and use it to simplify RLPDecode)

File: crates/common/types/receipt.rs (hunk 1 above)
Comment (fmoletta) [reply crates/common/types/receipt.rs]: Fixed!

File: crates/common/types/receipt.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (fmoletta) [reply crates/common/types/receipt.rs:143]: Update: Added the `decode_inner` method and simplified the logic of the `RLPDecode` impl

**Review by mpaulucci — APPROVED** [review]

**Pull request merged**